include/sparrow/buffer/dynamic_bitset/bitset_iterator.hpp

```diff
@@ -94,6 +94,10 @@ namespace sparrow
     {
         if constexpr (is_const)
         {
+            if(p_bitset->data() == nullptr)
```

Comment:
Nanoarrow don't create bitmap buffer if all elements are no null. But in our code we always create a dynamic_bitset, which can take as input a nullptr but until this fix, don't really support it.
In the case of const methods, I always return true.
For non const I assert. We should not reach this assert as we don't permit the modification of non sparrow arrays.